flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionCapacitySchedulerITCase.java

```diff
@@ -445,39 +443,45 @@ private static Map<String, String> getFlinkConfig(final String host, final int p
-                                null,
-                                RunTypes.YARN_SESSION,
-                                1);
-                    } catch (Exception e) {
-                        assertTrue(
-                                ExceptionUtils.findThrowableWithMessage(
-                                                e, "to unknown queue: doesntExist")
-                                        .isPresent());
-                    }
-                    assertThat(
-                            yarTestLoggerResource.getMessages(),
-                            hasItem(
-                                    containsString(
-                                            "The specified queue 'doesntExist' does not exist. Available queues")));
+                    assertThatThrownBy(
+                                    () ->
+                                            runWithArgs(
+                                                    new String[] {
+                                                        "-j",
+                                                        flinkUberjar.getAbsolutePath(),
+                                                        "-t",
+                                                        flinkLibFolder.getAbsolutePath(),
+                                                        "-jm",
+                                                        "768m",
+                                                        "-tm",
+                                                        "1024m",
+                                                        "-qu",
+                                                        "doesntExist"
+                                                    },
+                                                    "to unknown queue: doesntExist",
+                                                    null,
+                                                    RunTypes.YARN_SESSION,
+                                                    1))
+                            .isInstanceOf(Exception.class)
+                            .satisfies(
+                                    new Condition<Throwable>() {
+                                        @Override
+                                        public boolean matches(Throwable value) {
+                                            return ExceptionUtils.findThrowableWithMessage(
+                                                            value, "to unknown queue: doesntExist")
+                                                    .isPresent();
+                                        }
+                                    });
+
+                    assertThat(yarLoggerAuditingExtension.getMessages())
+                            .anyMatch(
+                                    s ->
+                                            s.contains(
+                                                    "The specified queue 'doesntExist' does not exist. Available queues"));
```

Comment:
```suggestion
                            .anySatisfies(
                                    s ->
                                            assertThat(s).contains(
                                                    "The specified queue 'doesntExist' does not exist. Available queues"));
```
This will give a better error message.